crates/anvil/src/eth/api.rs

```diff
@@ -1361,6 +1364,71 @@ impl EthApi {
         .map(U256::from)
     }
 
+    /// Fills a transaction request with default values for missing fields.
+    ///
+    /// This method populates missing transaction fields like nonce, gas limit,
+    /// chain ID, and fee parameters with appropriate defaults.
+    ///
+    /// Handler for ETH RPC call: `eth_fillTransaction`
+    pub async fn fill_transaction(
+        &self,
+        mut request: WithOtherFields<TransactionRequest>,
+    ) -> Result<FillTransactionResult<TypedTransaction>> {
+        node_info!("eth_fillTransaction");
+
+        let from = match request.as_ref().from() {
+            Some(from) => from,
+            None => self.accounts()?.first().copied().ok_or(BlockchainError::NoSignerAvailable)?,
+        };
+
+        let nonce = if let Some(nonce) = request.as_ref().nonce() {
+            nonce
+        } else {
+            self.request_nonce(&request, from).await?.0
+        };
+
+        if request.as_ref().has_eip4844_fields()
+            && request.as_ref().max_fee_per_blob_gas().is_none()
+        {
+            // Use the next block's blob base fee for better accuracy
+            let blob_fee = self.backend.fees().get_next_block_blob_base_fee_per_gas();
+            request.as_mut().set_max_fee_per_blob_gas(blob_fee);
+        }
+
+        if request.as_ref().blob_sidecar().is_some()
+            && request.as_ref().blob_versioned_hashes.is_none()
+        {
+            request.as_mut().populate_blob_hashes();
+        }
+
+        if request.as_ref().gas_limit().is_none() {
+            let estimated_gas = self
+                .estimate_gas(request.clone(), Some(BlockId::latest()), EvmOverrides::default())
+                .await?;
+            request.as_mut().set_gas_limit(estimated_gas.to());
+        }
+
+        if request.as_ref().gas_price().is_none() {
+            let tip = if let Some(tip) = request.as_ref().max_priority_fee_per_gas() {
+                tip
+            } else {
+                let tip = self.lowest_suggestion_tip();
+                request.as_mut().set_max_priority_fee_per_gas(tip);
+                tip
+            };
+            if request.as_ref().max_fee_per_gas().is_none() {
+                request.as_mut().set_max_fee_per_gas(self.gas_price() + tip);
+            }
+        }
+
+        let typed_tx = self.build_typed_tx_request(request, nonce)?;
+        let tx = self.sign_request(&from, typed_tx)?;
```

Comment:
this we can't use here, because this expects an actual signer, so this would currently fail for arbitrary senders

what we should do instead here is use the simulate signature to construct `build_typed_transaction` directly